lib/mayaUsd/ufe/MayaUsdContextOps.cpp

```diff
@@ -650,53 +714,106 @@ Ufe::UndoableCommand::Ptr MayaUsdContextOps::doOpCmd(const ItemPath& itemPath)
     } else if (itemPath[0] == kBindMaterialToSelectionItem) {
         std::shared_ptr<Ufe::CompositeUndoableCommand> compositeCmd;
         if (auto globalSn = Ufe::GlobalSelection::get()) {
             for (auto&& selItem : *globalSn) {
-                if (BindMaterialUndoableCommand::CompatiblePrim(selItem)) {
+                if (sceneItemSupportsShading(selItem)) {
                     if (!compositeCmd) {
                         compositeCmd = std::make_shared<Ufe::CompositeUndoableCommand>();
                     }
                     compositeCmd->append(std::make_shared<BindMaterialUndoableCommand>(
-                        selItem->path(), fItem->prim().GetPath()));
+                        selItem->path(), _item->prim().GetPath()));
                 }
             }
         }
         return compositeCmd;
     } else if (itemPath[0] == UnbindMaterialUndoableCommand::commandName) {
-        return std::make_shared<UnbindMaterialUndoableCommand>(fItem->path());
+        return std::make_shared<UnbindMaterialUndoableCommand>(_item->path());
 #ifdef UFE_V4_FEATURES_AVAILABLE
     } else if (itemPath.size() == 3u && itemPath[0] == kAssignNewMaterialItem) {
-        // Make a copy so that we don't change the user's original selection.
-        Ufe::Selection sceneItems(*Ufe::GlobalSelection::get());
-        // As per UX' wishes, we add the item that was right-clicked,
-        // regardless of its selection state.
-        sceneItems.append(fItem);
-        if (sceneItems.size() > 0u) {
-            return std::make_shared<InsertChildAndSelectCommand>(
-                UsdUndoAssignNewMaterialCommand::create(sceneItems, itemPath[2]));
-        }
+        // In single context item mode, only assign material to the context item.
+        return std::make_shared<InsertChildAndSelectCommand>(
+            UsdUndoAssignNewMaterialCommand::create(_item, itemPath[2]));
     } else if (itemPath.size() == 3u && itemPath[0] == kAddNewMaterialItem) {
         return std::make_shared<InsertChildAndSelectCommand>(
-            UsdUndoAddNewMaterialCommand::create(fItem, itemPath[2]));
+            UsdUndoAddNewMaterialCommand::create(_item, itemPath[2]));
     } else if (itemPath.size() == 3u && itemPath[0] == kAssignExistingMaterialItem) {
-        std::shared_ptr<Ufe::CompositeUndoableCommand> compositeCmd;
-        Ufe::Selection                                 sceneItems(*Ufe::GlobalSelection::get());
-        sceneItems.append(fItem);
-        for (auto& sceneItem : sceneItems) {
-            if (BindMaterialUndoableCommand::CompatiblePrim(sceneItem)) {
-                if (!compositeCmd) {
-                    compositeCmd = std::make_shared<Ufe::CompositeUndoableCommand>();
+        // In single context item mode, only assign material to the context item.
+        return std::make_shared<BindMaterialUndoableCommand>(_item->path(), SdfPath(itemPath[2]));
```

Comment:
According to new spec from Nat, only apply this command to single context item (not plus selection). New bulk menu is for selection.